Fabric.Identity.API/scripts/Register.ps1

```diff
@@ -235,12 +260,36 @@ function Add-PermissionToRole($authUrl, $roleId, $permission, $accessToken)
 
 function Add-RoleToGroup($authUrl, $groupName, $role, $accessToken)
 {
+    $role.permissions = $null
     $encodedGroupName = [System.Web.HttpUtility]::UrlEncode($groupName)
     $url = "$authUrl/groups/$encodedGroupName/roles"
-    $body = $role
+    $body = @($role)
     return Invoke-Post $url $body $accessToken
 }
 
+function Add-RoleToGroupSafe($authUrl, $groupName, $role, $accessToken){
+    try{
+        Write-Host "    Adding Role: $($role.name) to Group $groupName..."
+        Add-RoleToGroup -authUrl $authUrl -groupName $groupName -role $role -accessToken $accessToken
+    }catch{
+        $exception = $_.Exception
+        if($exception -ne $null -and $exception.Response -ne $null)
+        {
+            $error = Get-ErrorFromResponse -response $exception.Response
+            if($error.Contains("$($role.id) already exists")){
```

Comment:
Do we want this check or a 409 check in case error message changes?

Reply:
We have to first refactor the API to return a 409, currently it returns a 400 in this case unfortunately. We can open an issue in the Fabric.Authorization repo for that work, then we can update this.